index.js

```diff
@@ -121,12 +121,29 @@ class MiLightHubPlatform {
             platform.log('Using HTTP server at ' + platform.host);
           }
         }
-        for (var name in settings.group_id_aliases) {
-          var values = settings.group_id_aliases[name];
-          var lightInfo = { name: name, device_id: values[1], group_id: values[2], remote_type: values[0], uid: '0x' + values[1].toString(16).toUpperCase() + '/' + values[0] + '/' + values[2] };
-          lightList.push(lightInfo);
+        if(settings.group_id_aliases) {
+          for (var name in settings.group_id_aliases) {
+            var values = settings.group_id_aliases[name];
+            var lightInfo = { name: name, device_id: values[1], group_id: values[2], remote_type: values[0], uid: '0x' + values[1].toString(16).toUpperCase() + '/' + values[0] + '/' + values[2] };
+            lightList.push(lightInfo);
+          }
+          platform.syncLightLists(lightList);
+        } else {
+          const aliases_path = '/aliases';
+          platform.apiCall(aliases_path).then(response => {
+            if (response) {
+              const aliases = JSON.parse(response);
+              // TODO: check if we have to paginate (no API docs for that?)
+              platform.log(aliases);
+              for (var alias in aliases.aliases) {
+                //var lightInfo = { name: alias.alias, device_id: alias.device_id, group_id: alias.group_id, remote_type: alias.device_type, uid: '0x' + alias.device_id.toString(16).toUpperCase() + '/' + alias.device_type + '/' + alias.group_id };
+                var lightInfo = { name: alias.alias, device_id: alias.device_id, group_id: alias.group_id, remote_type: alias.device_type, uid: '0x' + alias.device_id + '/' + alias.device_type + '/' + alias.group_id };
```

Comment:
This will, eventually, throw TypeError later, because `alias` contains only index from table, not object.

Reply:
I don't have 1.11 installed locally as I don't have a free esp to test right now, the log output would help more. `aliases` has different content in different situations is what you're saying?

Reply:
I'm on 1.11.2 and `.../aliases` returns
```
{"aliases":[{"alias":"devicename","id":1,"device_id":123,"group_id":4,"device_type":"rgb_cct"}],"page":1,"count":1,"num_pages":1}
```

I've had to add simple fix after line 138 (`alias = aliases.aliases[alias];`) to make it working again.

Reply:
Ah, the iterator, darn, I'm jumping between languages too much ^^ Thanks, that should be an easy fix then. Now I just need to check if the whole pagination is real.